src/services/WaveformDataUtils.js

```diff
@@ -161,15 +165,55 @@ export default class WaveformDataUtils {
    * @param {Object} peaksInstance - current peaks instance for the waveform
    */
   activateSegment(id, peaksInstance) {
-    const segment = peaksInstance.segments.getSegment(id);
+    const validatedPeaks = this.initialSegmentValidation(id, peaksInstance);
+    const segment = validatedPeaks.segments.getSegment(id);
     segment.update({
       editable: true,
       color: COLOR_PALETTE[2]
     });
     let startTime = segment.startTime;
     // Move play head to the start time of the selected segment
-    peaksInstance.player.seek(startTime);
+    validatedPeaks.player.seek(startTime);
 
+    return validatedPeaks;
+  }
+
+  /**
+   * Adjust segment's end time to depict the valid time range for it to be spanned
+   * At initial load of the editor for all segments with invalid end times,
+   * segment.endTime is set to duration by default
+   * @param {String} id - ID of the segment being edited
+   * @param {Object} peaksInstance - current peaks instance for the waveform
+   */
+  initialSegmentValidation(id, peaksInstance) {
+    const segment = peaksInstance.segments.getSegment(id);
+    const duration = Math.round(peaksInstance.player.getDuration() * 100) / 100;
+    // Segments before and after the current segment
+    const { before, after } = this.findWrapperSegments(segment, peaksInstance);
+
+    // Check for margin of +/- 0.02 milliseconds to be considered
+    let isDuration = time => {
+      return time <= duration + 0.02 && time >= duration - 0.02;
+    };
+    if (
+      before &&
+      segment.startTime < before.endTime &&
+      !isDuration(before.endTime)
+    ) {
+      segment.update({ startTime: before.endTime });
+    } else if (isDuration(segment.endTime)) {
+      const allSegments = this.sortSegments(peaksInstance, 'startTime');
+      segment.update({
+        endTime: allSegments.filter(seg => seg.startTime > segment.startTime)[0]
+          .startTime
+      });
```

Comment:
Why is segment.endTime being set to a later segment's start time if it is within the 0.02ms margin of the duration?  Wouldn't that make the later segment be too small? Maybe this case is meant to be `!isDuration(segment.endTime)`?

Reply:
What this does is, when there is a timespan in the middle spanning towards the end of the file overlapping other segments after it, the endTime is set to the startTime of the next consecutive segment. 
Example: 
(segment editing) Segment 1: start = 00:02:00.00, end = 00:06.17.430 (end time of the file)
(later segment) Segment 2: start = 00:04:00.00, end = 00:05:00.000 
This code sets, `Segment 1.end = Segment 2.start` so that it doesn't go over Segment 2.
And it doesn't change Segment 2 (later segment).
The end result is;
Segment 1: start = 00:02:00.000 end = 00:04:00.000
Segment 2: start = 00:04:00.000 end = 00:05:00.000